src/devices/Mpu6886/Mpu6886.cs

```diff
@@ -0,0 +1,456 @@
+        /// </summary>
+        /// <param name="iterations">The number of sample gyroscope values to read</param>
+        /// <returns>The calulated offset vector</returns>
+        public Vector3 Calibrate(int iterations)
+        {
+            GyroscopeOffset = new Vector3(0, 0, 0);
+            Thread.Sleep(2);
+
+            var gyrSum = new double[3];
+
+            for (int i = 0; i < iterations; i++)
+            {
+                var gyr = GetRawGyroscope();
+
+                gyrSum[0] += gyr.X;
+                gyrSum[1] += gyr.Y;
+                gyrSum[2] += gyr.Z;
+
+                Thread.Sleep(2);
+            }
+
+            Vector3 offset = new Vector3((float)(gyrSum[0] / iterations), (float)(gyrSum[1] / iterations), (float)(gyrSum[2] / iterations));
+
+            GyroscopeOffset = offset;
+
+            return offset;
+        }
+
+        /// <summary>
+        /// Gets and sets the gyroscope offset in the GyroscopeOffsetAdjustment registers of the MPU6886.
+        /// Setting the offset can be usefull when a custom callibration calculation is used, instead of the Calibrate function of this class.
+        /// </summary>
+        public Vector3 GyroscopeOffset
+        {
+            get
+            {
+                Span<Byte> vec = stackalloc byte[6];
+                Read(Mpu6886.Register.GyroscopeOffsetAdjustmentXHighByte, vec);
+
+                Vector3 v = new Vector3();
+                v.X = (short)(vec[0] << 8 | vec[1]);
+                v.Y = (short)(vec[2] << 8 | vec[3]);
+                v.Z = (short)(vec[4] << 8 | vec[5]);
+
+                return v;
+            }
+
+            set
+            {
+                Span<Byte> registerAndOffset = stackalloc byte[7];
+                Span<Byte> offsetbyte = stackalloc byte[2];
+
+                registerAndOffset[0] = (byte)Mpu6886.Register.GyroscopeOffsetAdjustmentXHighByte;
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.X);
+                registerAndOffset[1] = offsetbyte[0];
+                registerAndOffset[2] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Y);
+                registerAndOffset[3] = offsetbyte[0];
+                registerAndOffset[4] = offsetbyte[1];
+
+                BinaryPrimitives.WriteInt16BigEndian(offsetbyte, (short)value.Z);
+                registerAndOffset[5] = offsetbyte[0];
+                registerAndOffset[6] = offsetbyte[1];
+
+                _i2c.Write(registerAndOffset);
+            }
+        }
+
+        /// <summary>
+        /// Reset the internal registers and restores the default settings. (Datasheet, page 47)
+        /// </summary>
+        public void Reset()
+        {
+            _i2c.Write(stackalloc byte[] { (byte)Mpu6886.Register.PowerManagement1, 0b1000_0000 });
+            Thread.Sleep(10);

```

Comment:
maybe add a general comment about the Thread.Sleep() that they are from the datasheet. If they are not from the datasheet, then the question is why those Thread.Sleep()?

Reply:
yes, according to datasheet startup times & delays. (e.g. p10)
added a general comment to explain